kratos/processes/simple_mortar_mapper_process.h

```diff
@@ -701,10 +701,14 @@ class KRATOS_API(KRATOS_CORE) SimpleMortarMapperProcess
         r_slave_geometry.PointLocalCoordinates(aux_coords, r_slave_geometry.Center());
         const array_1d<double, 3> slave_normal = r_slave_geometry.UnitNormal(aux_coords);
 
+        // The model part as const to avoid race conditions
+        const auto& r_const_origin_model_part = mOriginModelPart;
+        const auto& r_const_destination_model_part = mDestinationModelPart;
+
         for (auto it_pair = pIndexesPairs->begin(); it_pair != pIndexesPairs->end(); ++it_pair ) {
             const IndexType master_id = pIndexesPairs->GetId(it_pair); // MASTER
 
-            const auto& r_master_geometry = mOptions.Is(ORIGIN_SKIN_IS_CONDITION_BASED) ? mOriginModelPart.pGetCondition(master_id)->GetGeometry() : mOriginModelPart.pGetElement(master_id)->GetGeometry();
+            const auto& r_master_geometry = mOptions.Is(ORIGIN_SKIN_IS_CONDITION_BASED) ? r_const_origin_model_part.pGetCondition(master_id)->GetGeometry() : r_const_origin_model_part.pGetElement(master_id)->GetGeometry();
```

Comment:
@loumalouomega working with pGet entitites is very very slow due to the generation and destruction of the smart pointer in the loop. Why you need that?

Reply:
Huuum, if Get is faster I can try to change it, in fact I thought it was faster because I thought the PointerVectorSet stored Pointers and therefore dereferencing was not necessary

Reply:
I will try with the Get methods and revert those commits in this PR

Reply:
I see, but to my experience (see #7950) the creation of the temporal pointer is very costly. But the dereferencing is done in compile time.  (because you are not changing the reference counter)